app/code/core/Mage/Sales/Model/Resource/Quote.php

```diff
@@ -156,17 +156,20 @@ public function getReservedOrderId($quote)
     /**
      * Check is order increment id use in sales/order table
      *
-     * @param string $orderIncrementId
+     * @param string|integer $orderIncrementId
+     *
      * @return boolean
      */
     public function isOrderIncrementIdUsed($orderIncrementId)
     {
-        $adapter   = $this->_getReadAdapter();
-        $bind      = array(':increment_id' => $orderIncrementId);
-        $select    = $adapter->select();
-        $select->from($this->getTable('sales/order'), 'entity_id')
+
+        $adapter = $this->_getReadAdapter();
+        $fields = $adapter->describeTable($this->getTable('sales/order'));
+        $orderIncrementId = $adapter->prepareColumnValue($fields['increment_id'], $orderIncrementId);
```

Comment:
To make merging upstream changes easier commits should have minimal number of lines impacted. In this case this new line 168 could simply be added as the first or second line of the method and all of the other lines left unchanged.